R/config.R

```diff
@@ -1,45 +1,81 @@
-orderly_locate_config <- function() {
-  root <- find_file_descend("orderly_config.yml")
-  if (is.null(root)) {
-    stop("Reached root without finding 'orderly_config.yml'")
-  }
-  orderly_config$new(root)
-}
-
-
-orderly_config_get <- function(x, locate = FALSE) {
-  if (inherits(x, "orderly_config")) {
-    x
-  } else if (is.null(x) && locate) {
+##' Retrieve orderly config object.
+##'
+##' @inheritParams orderly_list
+##'
+##' @return An R6 object representing the orderly config.
+##' @export
+##'
+##' @examples
+##' # The orderly demo, with lots of potential reports:
+##' path <- orderly::orderly_example("demo")
+##'
+##' orderly::orderly_config(path)
+orderly_config <- function(root = NULL, locate = TRUE) {
+  if (inherits(root, "orderly_config")) {
+    root
+  } else if (is.null(root) && locate) {
     orderly_locate_config()
-  } else if (is.character(x)) {
-    orderly_config$new(x)
+  } else if (is.character(root)) {
+    orderly_config_$new(root)
   } else {
     stop("Invalid input")
   }
 }
 
+orderly_locate_config <- function() {
+  root <- find_file_descend("orderly_config.yml")
+  if (is.null(root)) {
+    stop("Reached root without finding 'orderly_config.yml'")
+  }
+  orderly_config_$new(root)
+}
 
-orderly_config <- R6::R6Class(
+##'
+##' @rdname orderly_config
+orderly_config_ <- R6::R6Class(
   "orderly_config",
 
```

Comment:
Can you add `cloneable = FALSE` to stop the noise in the docs about the clone method?